tiledb/sm/cpp_api/enumeration_experimental.h

```diff
@@ -223,6 +223,24 @@ class Enumeration {
   /*          STATIC FUNCTIONS         */
   /* ********************************* */
 
+  /**
+   * Create an empty enumeration.
+   *
+   * @param ctx The context to use.
+   * @param ordered Whether or not to consider this enumeration ordered.
+   * @param type The datatype of the enumeration values. This is automatically
+   *        deduced if not provided.
+   */
+  static Enumeration create(
```

Comment:
How about we call it `create_empty`?

Reply:
Done. And thanks for this! I was mulling over if I should pipe this through as an entirely separate API or not while also considering @KiterLuc's clue stick about the single empty string case and it really ended up helping clean up the constructor logic.